File: app/src/main/java/pl/elpassion/door/githubclient/service/github/GithubRepositoriesService.java
package pl.elpassion.door.githubclient.service.github;

import pl.elpassion.door.githubclient.service.github.response.RepositoriesSearchResponse;
import retrofit2.http.GET;
import retrofit2.http.Query;
import rx.Observable;

public interface GithubRepositoriesService {

    @GET("search/repositories")
    Observable<RepositoriesSearchResponse> searchForReposByName(@Query("q") String name);

}

File: app/src/main/java/pl/elpassion/door/githubclient/service/github/GithubRepositoriesService.kt
package pl.elpassion.door.githubclient.service.github

import pl.elpassion.door.githubclient.service.github.response.RepositoriesSearchResponse
import retrofit2.http.GET
import retrofit2.http.Query
import rx.Observable

interface GithubRepositoriesService {


    @GET("search/repositories")
    fun searchForReposByName(@Query("q") name: String): Observable<RepositoriesSearchResponse>

}